tests/neg/6571.scala

```diff
@@ -0,0 +1,28 @@
+object Test {
+  class Inv[T]
+  type M[t] = t match {
+    case Inv[u] => u
+  }
+  def foo: M[Inv[Int] & Inv[String]] = "" // error
+  //                                   ^^
```

Comment:
I think it would be good to add a check file that tests the actual error message.